src/Forms/GridField/GridFieldDataColumns.php

```diff
@@ -223,31 +223,6 @@ public function getColumnMetadata($gridField, $column)
         ];
     }
 
-    /**
-     * Translate a Object.RelationName.ColumnName $columnName into the value that ColumnName returns
-     *
-     * @param ModelData $record
-     * @param string $columnName
-     * @return string|null - returns null if it could not found a value
-     * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it.
-     */
-    protected function getValueFromRelation($record, $columnName)
-    {
-        Deprecation::notice('5.4.0', 'Will be removed without equivalent functionality to replace it.');
-        $fieldNameParts = explode('.', $columnName ?? '');
-        $tmpItem = clone($record);
-        for ($idx = 0; $idx < sizeof($fieldNameParts ?? []); $idx++) {
-            $methodName = $fieldNameParts[$idx];
-            // Last mmethod call from $columnName return what that method is returning
-            if ($idx == sizeof($fieldNameParts ?? []) - 1) {
-                return $tmpItem->XML_val($methodName);
```

Comment:
This was the only instance of `XML_val()` being used anywhere, and this method itself is never used, so easier to just remove it.